x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetMlAutoscalingStats.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.action;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.persistent.PersistentTasksCustomMetadata;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.ml.action.GetMlAutoscalingStats;
+import org.elasticsearch.xpack.core.ml.action.GetMlAutoscalingStats.Request;
+import org.elasticsearch.xpack.core.ml.action.GetMlAutoscalingStats.Response;
+import org.elasticsearch.xpack.ml.autoscaling.MlAutoscalingResourceTracker;
+import org.elasticsearch.xpack.ml.process.MlMemoryTracker;
+
+public class TransportGetMlAutoscalingStats extends TransportMasterNodeAction<Request, Response> {
+
+    private final Client client;
+    private final MlMemoryTracker mlMemoryTracker;
+
+    @Inject
+    public TransportGetMlAutoscalingStats(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Client client,
+        MlMemoryTracker mlMemoryTracker
+    ) {
+        super(
+            GetMlAutoscalingStats.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            Request::new,
+            indexNameExpressionResolver,
+            Response::new,
+            ThreadPool.Names.SAME
+        );
+        this.client = client;
+        this.mlMemoryTracker = mlMemoryTracker;
+    }
+
+    @Override
+    protected void masterOperation(Task task, Request request, ClusterState state, ActionListener<Response> listener) {
+
+        if (mlMemoryTracker.isRecentlyRefreshed()) {
+            MlAutoscalingResourceTracker.getMlAutoscalingStats(
+                state,
+                client,
+                request.timeout(),
+                mlMemoryTracker,
+                ActionListener.wrap(autoscalingResources -> listener.onResponse(new Response(autoscalingResources)), listener::onFailure)
+            );
+        } else {
+            mlMemoryTracker.refresh(
```

Comment:
I enforced the timeout and return a 408 in case. The outer API will make proper use of that and not return the ML metrics(but the other ones) and the autoscaler should interpret the ML-part as a no-op.

That way we still return something if it is quick or if the refresh was about to finish from a previous run.